src/pspc/iterator/FilmIteratorBase.h

```diff
@@ -0,0 +1,389 @@
+
+      /**
+      * Check whether or not the walls are athermal (this is only true
+      * if all of the values in the chi array are zero)
+      */
+      bool isAthermal() const;
+
+      /**
+      * Set the value of chi between species s and wall w
+      * 
+      * \param w  wall index, 0 or 1
+      * \param s  species index, 0 <= id < nVertex
+      * \param chi  value of chi(s,w)
+      */
+      void setChi(int s, int w, double chi);
+
+      /**
+      * Calculate value of phi, the overall volume fraction of unit cell occupied 
+      * by the wall.
+      * 
+      * Assumes that wall is well-defined (t_<T_, T_<L where L is unit cell length)
+      */
+      double const phi() const;
+
+      /**
+      * Write the concentration field of the walls to an output stream, in basis format.
+      * 
+      * \param out  output stream
+      */
+      void writeWallField(std::ostream &out) const;
+
+      /**
+      * Write the concentration field of the walls to a file specified by 'filename', 
+      * in basis format.
+      * 
+      * \param filename  name of file to which data will be written
+      */
+      void writeWallField(std::string filename) const;
+
+      /**
+      * Write the concentration field of the walls to an output stream, in r-grid format.
+      * 
+      * \param out  output stream
+      */
+      void writeWallFieldRGrid(std::ostream &out) const;
+
+      /**
+      * Write the concentration field of the walls to a file specified by 'filename', 
+      * in r-grid format.
+      * 
+      * \param filename  name of file to which data will be written
+      */
+      void writeWallFieldRGrid(std::string filename) const;
+
+      /**
+      * Get the const DArray containing the concentration field for the walls by 
+      * const reference
+      */
+      DArray<double> const & wallCField() const;
+
+      /**
+      * Get value of normalVec, the lattice basis vector that is
+      * normal to the walls (either 0, 1, or 2 in 3D)
+      */
+      int normalVec();
+
+      /**
+      * Get const value of normalVec
+      */
+      int const normalVec() const;
+
+      /**
+      * Get value of interfaceThickness
+      */
+      double interfaceThickness();
+
+      /**
+      * Get const value of interfaceThickness
+      */
+      double const interfaceThickness() const;
+
+      /**
+      * Get value of wallThickness
+      */
+      double wallThickness();
+
+      /**
+      * Get const value of wallThickness
+      */
+      double const wallThickness() const;
+
+      /**
+      * Get chi matrix by reference
+      */
+      DMatrix<double>& chi();
+
+      /**
+      * Get const chi matrix by reference
+      */
+      DMatrix<double> const & chi() const;
+
+      /**
+      * Get the chi parameter between wall w and species s by reference
+      * 
+      * \param s  species index, 0 <= id < nVertex
+      * \param w  wall index, 0 or 1
+      */
+      double& chi(int s, int w);
```

Comment:
Why return a single value by non-const reference. Did you intend to provide element by element write access?

Reply:
Oh, this is a base class, and it's protected:  I see that you probably did mean to provide write access to subclasses.

Reply:
I'm glad you brought this up, because I added a setter function setChi() that is used for write access to the chi matrix, so we don't actually need this non-const function.